### ISSUE TITLE
When used with react strict mode, the realtime database does not subscribe properly.

### ISSUE THREAD

**Issue by Jannchie** [supabase/supabase#7799]
# Bug report

## Describe the bug

When we develop with the React strict mode, we can't build subscriptions correctly.

## To Reproduce

Turn on strict mode:

```
ReactDOM.createRoot(document.getElementById("root") as HTMLElement).render(
  <React.StrictMode>
    <App />
  </React.StrictMode>
);
```

Using subscriptions in components:

``` tsx
function App() {
  useEffect(() => {
    const subs = supabase
      .from("objects")
      .on("UPDATE", (payload) => {
        dispatch(updateGameObject(payload.new.data));
      })
      .subscribe(console.log);
    subs.onClose(() => {
      console.log("on closed");
      supabase.removeAllSubscriptions();
    });
    return () => {
      subs.unsubscribe();
    };
  }, []);
  return <div />
}
```

## Expected behavior

In strict mode, useEffect is triggered twice. In theory, it should quickly establish and close a connection once, and then establish it again.

However, in practice, it is not possible to subscribe properly.

## Screenshots

The output in the console is as follows:

![image](https://user-images.githubusercontent.com/29743310/179883362-61ef1a3a-8baf-4659-81bf-ec22fa85d3d9.png)

## System information

- OS: Windows
- Browser (if applies): Edge
- Version of supabase-js: ^1.35.4
- Version of Node.js: 16
- Version of react: 18

## Additional context

In #7771 it is suggested that turning on strict mode can cause problems.

**Comment by kristianeboe** [comment]
Is this with React 18? React 18's Strict mode has been nothing but trouble for me since it started running useEffect twice in localhost

**Comment by Jannchie** [comment]
Yes, I'm using React 18. The above code should work fine if useEffect is called twice. But in reality, it does not.

**Comment by stevharve** [comment]
Same issue here, Vite with React 18 and a brand new project. Strict Mode is enabled.

```tsx
useEffect(() => {
    const sub = supabase
      .from('message')
      .on('*', (message) => {
        console.log('New Message');
        handleNewMessage(message.new);
      })
      .subscribe();
    return () => {
      supabase.removeSubscription(sub);
    };
  }, []);
```

**Comment by kiwicopple** [comment]
got it! Sending this to the realtime-js repo 👍
